scintillometry/io/psrfits/hdu.py

```diff
@@ -135,7 +216,7 @@ class SubintHDU(HDUWrapper):
     _sample_shape_maker = namedtuple('SampleShape', 'nbin, nchan, npol')
     _shape_maker = namedtuple('Shape', 'nsample, nbin, nchan, npol')
 
-    def __new__(cls, hdu, primary_hdu, verify=True):
+    def __new__(cls, primary_hdu, hdu=None, verify=True):
```

Comment:
Not sure I find this switch back more logical: the `Subint` surely is primarily concerned with its *own* hdu, it just needs the primary one for additional information. I think it is fine for the creation of a subint hdu to read `SubintHDU(primary_hdu=...)`